### ISSUE TITLE
Redundant condition in model.py

### ISSUE THREAD

**Issue by John1231983** [zhengyang-wang/Deeplab-v2--ResNet-101--Tensorflow#26]
I have check your code and your  note

>'is_training' argument is removed and 'self._batch_norm' changes. Basically, for a small batch size, it is better to keep the statistics of the BN layers (running means and variances) frozen, and to not update the values provided by the pre-trained model by setting 'is_training=False'. Note that is_training=False still updates BN parameters gamma (scale) and beta (offset) if they are presented in var_list of the optimiser definition. Set 'trainable=False' in BN fuctions to remove them from trainable_variables

It means that when we frozen BN, we will keep its parameters fixed **(including mean, variance, beta, gamma)**. in your bn layer in network.py, you have set `trainable=False`, hence` gamma` and `beta ` will not appear in list. So, I think you have a redundant condition in the lines of model.py

```
decoder_w_trainable = [v for v in decoder_trainable if 'weights' in v.name or 'gamma' in v.name] # lr * 10.0
decoder_b_trainable = [v for v in decoder_trainable if 'biases' in v.name or 'beta' in v.name] # lr * 20.0
		
```
Am I right? Or for a good condition, it must be `'gamma' not in v name`, instead of `'gamma' in v.name`

**Comment by zhengyang-wang** [comment]
Yes, currently it's redundant, but not because we fix BN. It's due to the fact that current models have no BN layer in the decoder. Note that we only fix BN for the encoder, where we do not train the BN parameters transferred from other models.

However, as introduced in README,

> Add 'phase' argument in network.py for future development. 'phase=True' means training. It is mainly for controlling batch normalization (if any) in the non-pre-trained part.

What you mention here is for the same purpose. I put it in this way in case I add a BN in the decoder part.
